File: app/src/main/java/com/mapmyindia/sdk/demo/java/adapter/NearByAdapter.java
package com.mapmyindia.sdk.demo.java.adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.mapmyindia.sdk.demo.R;
import com.mmi.services.api.nearby.model.NearbyAtlasResult;

import java.util.ArrayList;

/**
 * Created by CEINFO on 26-02-2019.
 */

public class NearByAdapter extends RecyclerView.Adapter<NearByAdapter.NearByView> {

    ArrayList<NearbyAtlasResult> list;

    public NearByAdapter(ArrayList<NearbyAtlasResult> list) {
        this.list = list;
    }

    @NonNull
    @Override
    public NearByView onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.auto_suggest_adapter_row, parent, false);
        return new NearByView(v);
    }

    @Override
    public void onBindViewHolder(@NonNull NearByView holder, int position) {
        holder.viewName.setText(list.get(position).getPlaceAddress());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class NearByView extends RecyclerView.ViewHolder {
        TextView viewName;

        public NearByView(View itemView) {
            super(itemView);
            viewName = itemView.findViewById(R.id.textView);
        }
    }
}

File: app/src/main/java/com/mapmyindia/sdk/demo/kotlin/adapter/NearByAdapter.kt
package com.mapmyindia.sdk.demo.kotlin.kotlin.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.mapmyindia.sdk.demo.R
import com.mmi.services.api.nearby.model.NearbyAtlasResult
import java.util.*

/**
 * Created by CEINFO on 26-02-2019.
 */
class NearByAdapter(internal var list: ArrayList<NearbyAtlasResult>) : androidx.recyclerview.widget.RecyclerView.Adapter<NearByAdapter.NearByView>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NearByView {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.auto_suggest_adapter_row, parent, false)
        return NearByView(v)
    }

    override fun onBindViewHolder(holder: NearByView, position: Int) {
        holder.viewName.text = list[position].placeAddress
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class NearByView(itemView: View) : androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView) {
        var viewName: TextView

        init {
            viewName = itemView.findViewById(R.id.textView)
        }
    }
}
